torch/_subclasses/functional_tensor.py

```diff
@@ -471,18 +477,23 @@ def mark_mutation_hidden_from_autograd(self, tensor) -> None:
 
 
 class PythonFunctionalizeAPI(BaseFunctionalizeAPI):
+    def __init__(self, mode):
+        super().__init__()
+        self.mode = mode
```

Comment:
remind me why we need to add this as an extra field, instead of just using `_detect_functional_mode()` everywhere we previously mocked up a fresh functional mode?

Reply:
The reason is in HOO, we temporarily pop the FunctionalMode, as a result, there is no global mode anymore.